### PR TITLE
CLI refactoring for testability - part 8

### PR DESCRIPTION
**Pull request by afoxman**
### Description

In part 8, I've added a `createDirectory` function to the Node tools package. I used it in the metro bundling code to replace calls to `fs.mkdirSync(...)`.

### Test plan

New tests for `createDirectory` pass. Existing build, test, bundle CI loop passes.

### CODE REVIEW

File: packages/tools-node/test/fs.test.ts [hunk 1]
```diff
@@ -28,4 +42,19 @@ describe("Node > FS", () => {
...
+    createDirectory(p);
+    expect(fs.existsSync(p)).toBeTrue();
+  });
+
+  test("createDirectory() creates a parent directory", () => {
+    const parent = path.join(testTempDir, "parentdir");
+    const p = path.join(parent, "testdir");
+    expect(fs.existsSync(parent)).toBeFalse();
+    createDirectory(p);
+    expect(fs.existsSync(parent)).toBeTrue();
```
Comment (tido64) [inline packages/tools-node/test/fs.test.ts:58]: Can we also verify that its mode/permissions are correctly set?

**Review by tido64 — APPROVED** [review]

File: packages/tools-node/test/fs.test.ts (hunk 1 above)
Comment (afoxman) [reply packages/tools-node/test/fs.test.ts:58]: I'll give it a shot. Who knows how well it will translate on windows :)

Comment (tido64) [reply packages/tools-node/test/fs.test.ts:58]: Making it conditional on unix filesystems is probably fine :)

**Review by tido64 — APPROVED** [review]

File: packages/tools-node/test/fs.test.ts (hunk 1 above)
Comment (afoxman) [reply packages/tools-node/test/fs.test.ts:68]: I found common ground that I think will work -- i feed in rwx|rx|rx and on macos i get that back plus a higher bit (which i ignore). On windows I get back rw|r|r which is good enough of a test, so I've made that the lowest common denominator here. I test for rw|r|r as well as "isDirectory". I'm going assume the exec bits will be there if rw|r|r is there.

Comment (afoxman) [reply packages/tools-node/test/fs.test.ts:68]: At some point I'm testing nodejs, and not the library code, so this seems good enough.

Comment (tido64) [reply packages/tools-node/test/fs.test.ts:68]: It doesn't check that we didn't accidentally set `rw|rw|rw` though. We want to make sure that only the owner has write permissions.

**Pull request merged**